source/routing/raptor.cpp

```diff
@@ -705,11 +808,44 @@ void RAPTOR::raptor_loop(Visitor visitor, const nt::RTLevel rt_level, uint32_t m
                     // We try to get on a vehicle, if we were already on a vehicle, but we arrived
                     // before on the previous via a connection, we try to catch a vehicle leaving this
                     // journey pattern point before
+
+                    // if we cannot board at this stop point, nothing to do
+                    if (!prec_labels.transfer_is_initialized(jpp.sp_idx) || !valid_stop_points[jpp.sp_idx.val]) {
+                        continue;
+                    }
+
+                    /// the time at which we arrive at stop point jpp.sp_idx (using at most count-1 transfers)
+                    //  hence we can board any vehicle arriving after previous_dt
                     const DateTime previous_dt = prec_labels.dt_transfer(jpp.sp_idx);
-                    if (prec_labels.transfer_is_initialized(jpp.sp_idx) && valid_stop_points[jpp.sp_idx.val]
-                        && (!is_onboard || visitor.better_or_equal(previous_dt, base_dt, *it_st))) {
-                        const auto tmp_st_dt =
-                            next_st->next_stop_time(visitor.stop_event(), jpp.idx, previous_dt, visitor.clockwise());
+                    const DateTime previous_walking_duration = prec_labels.walking_duration_transfer(jpp.sp_idx);
+
+                    /// we are at stop point jpp.idx at time previous_dt
```

Comment:
can you test `previous_walking_duration <= working_walking_duration` here and `continue` if false? 

it may probably avoid the binary search in `previous_walking_duration <= working_walking_duration`?